javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/HelperReferenceWrapper.java

```diff
@@ -161,6 +200,21 @@ private Method toMethod(Reference.Method method) {
             method.getName(),
             method.getDescriptor());
       }
+
+      @Override
+      public Stream<Field> getFields() {
+        return reference.getFields().stream()
+            .filter(this::isDeclaredAndNotPrivate)
```

Comment:
Why are private fields special? An optimization? Nestmates can access private fields within nest, would that be an issue?

Reply:
It's an optimization. This field check is mostly meant to detect changed/renamed fields when we're extending library classes - nested classes are not a valid scenario in this case. And if we use nested classes in our own instrumentation code the Java compiler will pick up all inconsistencies.